crates/jarscanner/src/lib.rs

```diff
@@ -38,27 +42,32 @@ fn file_name_to_class_names(file_name: &str) -> Result<Vec<String>, FileNameErro
             .replace(r"/", ".");
```

Comment:
These are regex right? I wonder if this is becoming the hotspot.

Reply:
A potential optimization might be to use regex crate, and move the regex compilation out of the loop, so we perform it once per expression?

Reply:
That's a good idea, I'll give it a go

Reply:
Take a look at my latest commit, these are the results:
```
---- Timing for 50 runs ----
        4.37 real         1.26 user         2.91 sys
Git Commit: 75e510346d1ab50b53003cc1eb6e4b247532bdf3
---------------------------
---- Timing for 50 runs ----
        4.26 real         1.24 user         2.88 sys
Git Commit: 75e510346d1ab50b53003cc1eb6e4b247532bdf3
---------------------------
---- Timing for 50 runs ----
        4.27 real         1.25 user         2.90 sys
Git Commit: 75e510346d1ab50b53003cc1eb6e4b247532bdf3
---------------------------
---- Timing for 50 runs ----
        4.21 real         1.22 user         2.89 sys
Git Commit: fd73ac001898a17ede1e584d98301868dc877160
---------------------------
---- Timing for 50 runs ----
        4.22 real         1.22 user         2.89 sys
Git Commit: fd73ac001898a17ede1e584d98301868dc877160
---------------------------
---- Timing for 50 runs ----
        4.20 real         1.21 user         2.87 sys
Git Commit: fd73ac001898a17ede1e584d98301868dc877160
---------------------------
---- Timing for 50 runs ----
        4.58 real         1.28 user         3.09 sys
Git Commit: 30ffea59861f4c8db85e86ea4131b78b5bf9b8c0
---------------------------
---- Timing for 50 runs ----
        4.52 real         1.28 user         3.10 sys
Git Commit: 30ffea59861f4c8db85e86ea4131b78b5bf9b8c0
---------------------------
---- Timing for 50 runs ----
        4.53 real         1.29 user         3.11 sys
Git Commit: 30ffea59861f4c8db85e86ea4131b78b5bf9b8c0
---------------------------
``` 

Honestly quite surprised here.